search/search_quality/assessment_tool/edits.hpp

```diff
@@ -38,18 +64,19 @@ class Edits
   // |relevance| differs from the original one.
   bool SetRelevance(size_t index, Relevance relevance);
 
-  std::vector<Relevance> const & GetRelevances() { return m_currRelevances; }
+  std::vector<Relevance> const & GetRelevances() const { return m_currRelevances; }
 
   void Clear();
   bool HasChanges() const;
+  bool HasChanges(size_t index) const;
 
 private:
   template <typename Fn>
-  typename std::result_of<Fn()>::type WithObserver(Fn && fn)
+  typename std::result_of<Fn()>::type WithObserver(Update const & update, Fn && fn)
   {
-    MY_SCOPE_GUARD(cleanup, [this]() {
+    MY_SCOPE_GUARD(cleanup, [&]() {
```

Comment:
The problem is with macro expansion and lambda capture list. I want to capture this and update (by reference), but it's not possible without helper surrounding parenthesis.